### PR TITLE
Botany framework additions!

### PR DESCRIPTION
**Pull request by Avatharian**
[FEATURE]
## About the PR 
This PR adds some new frameworks to Botany that allow for some new and useful features for plant customization and harvesting. It adds:

- The ability to make certain plants require specific reagents to be present in the pot in order to grow
- The ability to make certain plants require the use of specific tools to harvest them, based on tool flags (`TOOL_CUTTING,TOOL_SAWING`, etc) and custom messages based thereupon.
- Action bars! Using the harvest proc, plants can now support action bars, making harvesting take a certain amount of time, during which it can be interrupted like any other action bar.
- 2 variables which ease customization of plants; One makes it so the sprite of a particular plant does not grow with potency, the other disables the ability to extract seeds from the plant. 

Right now there are no plants in this PR that make use of the new features yet, but it lays the groundwork for a couple that are being worked on, and for other future additions and customization of existing plants.

## Why's this needed? 

We wanted to add some _interesting_ new plants but needed some new customization options for balance and flavor. I think that in general more features in botany that allow for more variation and flavor when it comes to growing various plants is a good thing. Currently with a _few_ notable exceptions, there isn't a lot of difference when it comes to growing one plant vs. another in the actual pot.

### CODE REVIEW

**Comment by Tiggersaurus** [comment]
Your friendly neighborhood pain in the ass Borg here, just checking how these future additions would affect (if at all) Borg botany! 💜

**Comment by Avatharian** [comment]
I can confirm that Borgs are still able to use the Omnitool to harvest plants that require tools. It's based on the tool flag rather than a specific whitelist. I don't believe there is anything else that could have possibly interfered with Borgany.

**Comment by Avatharian** [comment]
But thinking about Borgs and botany _did_ give me an idea for a new plant that could be useful to our robo-friends!

File: code/obj/machinery/plantpot.dm [hunk 1]
```diff
@@ -99,6 +99,7 @@
 		// We have this here as a check for whether or not the plant needs to update its sprite.
 		// Originally plantpots updated constantly but this was found to be rather expensive, so
 		// now it only does that if it needs to.
+	var/actionpassed //variable for action bar harvesting yay :D 0 = no action, 1 = action passed, 2 = action cancelled
```
Comment (ZeWaka) [inline code/obj/machinery/plantpot.dm:102]: please make use of #defines here for the values

Comment (Avatharian) [reply code/obj/machinery/plantpot.dm:102]: Sorry for the delay, I'll figure out how to do that!

Comment (Avatharian) [reply code/obj/machinery/plantpot.dm]: Alright, I think I figure it out.

**Pull request merged**